cmd/minishift/cmd/delete.go

```diff
@@ -46,23 +48,41 @@ func runDelete(cmd *cobra.Command, args []string) {
 
 	util.ExitIfUndefined(api, constants.MachineName)
 
+	if err := cluster.UnRegister(api, minishiftConfig.InstanceConfig); err != nil {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Error unregistering the VM: %s", err.Error()))
+	}
+
+	if minishiftConfig.InstanceConfig.IsRegister {
```

Comment:
Here is what I mean:

    func runDelete(cmd *cobra.Command, args []string) {
    	if minishiftConfig.InstanceConfig.IsRegistered {
    		err := cluster.UnRegister(api)
    		if err != nil {
    			handleFailedUnRegistration()
    		} 
    		instanceConfig.IsRegister = false
    		// not sure whether we need to handle failure, since we are going to delete the
    		// instance config altogther a little bit later on!?
     		instanceConfig.Write()
    	}
    }
    
    func handleFailedUnRegistration() {
    		userConfirmation := pkgUtil.ReadInputFromStdin("Current Minishift instance is registered     with subscription, " +
    		"Still want to Delete VM (y/n)")
    		if strings.ToUpper(userConfirmation) != "Y" {
    			atexit.ExitWithMessage(0, fmt.Sprintln("Delete aborted by User"))
    		}	
    }
    
Might need some tuning, but I think something like this makes the intend and what's actually happening much clearer.